test/e2e-olm/elasticsearch_test.go

```diff
@@ -512,3 +512,78 @@ func invalidMasterCountTest(t *testing.T) {
 	e2eutil.WaitForDeletion(t, f.Client.Client, cr, cleanupRetryInterval, cleanupTimeout)
 	t.Log("Finished successfully")
 }
+
+func nonDefaultStorageClassTest(t *testing.T) {
+	f := test.Global
+
+	ctx := test.NewContext(t)
+	namespace, err := ctx.GetWatchNamespace()
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("Found namespace: %v", namespace)
+
+	esUUID := utils.GenerateUUID()
+	t.Logf("Using UUID for elasticsearch CR: %v", esUUID)
+
+	if err = createElasticsearchSecret(t, f, ctx, esUUID); err != nil {
+		t.Fatal(err)
+	}
+
+	dataUUID := utils.GenerateUUID()
+	t.Logf("Using GenUUID for data nodes: %v", dataUUID)
+
+	// Create CR with a single node using ephemeral storage
+	nonDefaultStorageClassName := "gp2"
+	cr, err := createElasticsearchCR(t, f, ctx, esUUID, dataUUID, 1, PersistentStorageClassType, &nonDefaultStorageClassName)
+	if err != nil {
+		t.Fatalf("could not create exampleElasticsearch: %v", err)
+	}
+
+	dplName := fmt.Sprintf("elasticsearch-%v-cdm-%v-1", esUUID, dataUUID)
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, dplName, 1, retryInterval, timeout)
+	if err != nil {
+		t.Fatalf("timed out waiting for first node deployment %v: %v", dplName, err)
+	}
+
+	ctx.Cleanup()
+	e2eutil.WaitForDeletion(t, f.Client.Client, cr, cleanupRetryInterval, cleanupTimeout)
+	t.Log("Finished successfully")
+}
+
+func ephemeralStorageTest(t *testing.T) {
```

Comment:
this does not need to be an e2e test, a unit test should be enough to verify this logic

Reply:
The E2E test is removed.